src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs

```diff
@@ -12,64 +12,64 @@
 using Microsoft.CodeAnalysis.Editor;
 using Microsoft.CodeAnalysis.Editor.FindUsages;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.SignatureHelp;
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService;
 using Microsoft.VisualStudio.LanguageServices.LiveShare.CustomProtocol;
+using Microsoft.VisualStudio.LanguageServices.LiveShare.Protocol;
 using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.VisualStudio.LanguageServices.LiveShare
 {
     [ExportLspRequestHandler(LiveShareConstants.TypeScriptContractName, Methods.TextDocumentCompletionName)]
-    internal class TypeScriptCompletionHandlerShim : AbstractLiveShareHandlerShim<CompletionParams, object>
+    internal class TypeScriptCompletionHandlerShim : CompletionHandler, ILspRequestHandler<object, object?, Solution>
     {
-        [ImportingConstructor]
-        public TypeScriptCompletionHandlerShim([ImportMany] IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers)
-            : base(requestHandlers, Methods.TextDocumentCompletionName)
-        {
+        public async Task<object?> HandleAsync(object input, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+        {
+            // The VS LSP client supports streaming using IProgress<T> on various requests.
+            // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+            var request = ((JObject)input).ToObject<CompletionParams>(InProcLanguageServer.JsonSerializer);
```

Comment:
how do you know it's a JObject?

Reply:
If we don't specify a non-object type for liveshare to use to deserialize the input, it will leave it as JObject for us to deserialize.

Reply:
Can we specify JObject as the parameter instead?